Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -1775,49 +1786,199 @@ EntityRecord UpdateInstancedEntity(TransformData data)
+                default:
+                    break;
+            }
+
             return infoRecord;
         }
 
+        private void UpdateInstanceInfo_CopiesAndPrefabs(
+            InstanceInfoData data,
+            InstanceInfoRecord infoRecord,
+            EntityRecord instancedEntityRecord,
+            GameObject instanceRendererParent) {
+            // Make sure the other renderer is removed if this was not a copy or prefab before:
+            if (infoRecord.renderer != null) {
+                DestroyImmediate(infoRecord.renderer);
+                infoRecord.renderer = null;
+            }
+            infoRecord.DeleteInstanceObjects();
+
+            instanceNames.Add(infoRecord.go.name);
+
+            MeshSyncPlayerConfig config = GetConfigV();
+
+            if (!instanceNames.Contains(infoRecord.go.transform.parent.name)) {
+                bool visible = instancedEntityRecord.visibility.visibleInRender;
+
+                GameObject instanceObjectOriginal;
+                if (InstanceHandling == InstanceHandlingType.Copies) {
+                    instanceObjectOriginal = infoRecord.go;
+                }
+                else {
+                    instanceObjectOriginal = GetOrCreatePrefab(data, infoRecord);
+                }
+
+                if (instanceObjectOriginal == null) {
+                    return;
+                }
+
+                foreach (var mat in data.transforms) {
+                    GameObject instancedCopy;
+                    if (InstanceHandling == InstanceHandlingType.Copies) {
+                        instancedCopy = Instantiate(instanceObjectOriginal, instanceRendererParent.transform);
+                    }
+                    else {
+#if UNITY_EDITOR
+                        instancedCopy = (GameObject)PrefabUtility.InstantiatePrefab(instanceObjectOriginal,
+                        instanceRendererParent.transform);
+#else
+                        instancedCopy = Instantiate(instanceObjectOriginal, instanceRendererParent.transform);
+#endif
+                    }
+
+                    SetInstanceTransform(instancedCopy, instanceObjectOriginal, mat);
+
+                    if (config.SyncVisibility &&
+                        instancedEntityRecord.hasVisibility) {
+                        instancedCopy.SetActive(visible);
+                    }
+
+                    infoRecord.instanceObjects.Add(instancedCopy);
+                }
+            }
+        }
+
+        private static void SetInstanceTransform(GameObject instancedCopy, GameObject instanceObjectOriginal, Matrix4x4 mat) {
+            Transform objTransform = instancedCopy.transform;
+
+            var newMat = instanceObjectOriginal.transform.localToWorldMatrix * mat;
+
+            objTransform.localScale = newMat.lossyScale;
+            objTransform.position   = newMat.MultiplyPoint(Vector3.zero);
+
+            // Calculate rotation here to avoid gimbal lock issue:
+            Vector3 forward;
```

Comment:
Probably but this is faster

Reply:
Really ? This is fine, but I'd like to know how you measured it.
In my understanding, GetColumn() or GetRow() should be faster because it's a vector operation, and because it's a native API, rather than C# script.

Reply:
Bumping up